table/tables/tables.go

```diff
@@ -432,6 +433,12 @@ func (t *tableCommon) AddRecord(ctx sessionctx.Context, r []types.Datum, skipHan
 		if err != nil {
 			return 0, errors.Trace(err)
 		}
+	} else if hasExplicitRowID {
+		// if _tidb_rowid is explicitly assigned, we need to rebase the allocator
+		err = t.RebaseAutoID(ctx, recordID, false)
```

Comment:
True (it is also explained in the last meeting). So most of the caution outlined before still applies (e.g. there should only be 1 writer to the table at a time).

This is still useful in a single-writer use case though. For instance, if we do a full-restore via Loader, and the source data contains the `_tidb_rowid` column, previously we'll need to parse the SQL dump locally to get the `MAX(_tidb_rowid)` for rebase (`ALTER TABLE AUTO_INCREMENT`), but now this will be handled on the server side.